### DISCUSSION TITLE
How to parse params in the query event?

### DISCUSSION THREAD

**Discussion by tmtron** [prisma/prisma#14158] [Q&A] [answered]
In the [query event](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#event-based-logging) we get the `query` which can include `?` placeholders and the `params` which is an array-like string: e.g. `[\"abc2\",2022-07-05 14:18:03.680719400 UTC,2]`

We'd like to write a convenience function to replace the `?` with the actual param values - just for readability of the log-messages during development (It can be quite difficult to understand a long query with multiple parameters)

I tried to write a simple parsing logic for the params, but I think, currently there is no reliable way to parse that string (i.e. to get the individual items), because quotes and commas within a string are not escaped
- e.g. when the first parameter is `a "," in a string`, and the 2nd parameter is `p2` the `params` array is `[\"a \",\" in a string\",\"p2\"]`
- from the number of question marks in the query, we know that there must be 2 parameters , but it's not possible to know if these are
  1.  `a "," in a string` and `p2`
  2. or `a "` and `" in a string","p2"`

Notes:
* we use SQLite
* Maybe, there there is already a function to do that?
* Or is there a way to tell prisma to JSON encode the `params`?

**Comment by pantharshit00** [comment]
Hello @tmtron, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

I will suggest to open a feature request about this: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Comment by tmtron** [comment, marked as answer]
@pantharshit00 Thanks for the answer. Before opening a feature request, I'd like to hear thoughts of the prisma team on how to resolve this issue. 
My opinion is that the current implementation has a bug and that double quotes in parameters need to be escaped so that we can parse the `params` string correctly. 
A better approach would bet to change `params` to an array (instead of a string) - but this might break existing code.

**Reply by pantharshit00** [reply]
Current implementation for displaying parameter values is here: https://github.com/prisma/quaint/blob/657a56a38f6d61533ecbc71935eff61fbcf6af54/src/ast/values.rs#L110

As you can see only few values are quoted so this is a deliberate choice. So I will recommend opening a feature request and we can change this behaviour.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
